// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package helloworld;

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}

  rpc siguienteFecha (Fecha) returns (Fecha) {}

  rpc sumaDiasFecha (SumaFecha) returns (Fecha) {}

  rpc dimeCentroDiana (Vacio) returns (Centro) {}

  rpc disparaCannon (Disparo) returns (Tiro) {}

  rpc mejorDisparo (Vacio) returns (Mejor) {}

  rpc SayHelloAgain (HelloRequest) returns (HelloReply){}

  rpc SayHelloStreamReply (HelloRequest) returns (stream HelloReply) {}

  rpc SayHelloBidiStream (stream HelloRequest) returns (stream HelloReply) {}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}

message Fecha {
  int32 dia=1;
  int32 mes=2;
  int32 year=3;
}

message SumaFecha {
  int32 dia=1;
  int32 mes=2;
  int32 year=3;
  int32 num_dias=4;
}

message Centro {
  float distancia=1;
}

message Disparo {
  string nombre=1;
  float angulo=2;
  float velocidad=3;
}

message Tiro {
  float distancia=1;
}

message Mejor {
  string nombre=1;
  float distancia=2;
}

message Vacio {}
